main.go

```diff
@@ -1,9 +1,12 @@
 package main
 
 import (
+	"context"
+
 	"go.k6.io/k6/cmd"
+	"go.k6.io/k6/cmd/state"
 )
 
 func main() {
-	cmd.Execute()
+	cmd.Execute(state.NewGlobalState(context.Background()))
```

Comment:
So this change breaks xk6. It's a simple fix, and I'll push the change later today.

I needed to do this because tests in `cmd/tests` must call an exported interface from `cmd` to actually execute the command. The way they did it while they were in `cmd` is with `newRootCommand(ts.GlobalState).execute()`, but exporting that involved exporting a whole lot of related types and methods. So it was easier to just change the signature of this already exported `cmd.Execute()` function.

Reply:
This is fixed by https://github.com/grafana/xk6/commit/52e45894f2d4509d8c877fe4af9289d52b5ed17d and and3ebefcebc, but merging these changes would still break building k6 binaries with versions prior to this change.

So I think we should either avoid breaking this, or handle the failure in xk6 in a more sophisticated way. E.g. try building with both `cmd.Execute()` and `cmd.Execute(state.NewGlobalState(context.Background()))`?

WDYT?

Reply:
1. add `ExecuteWIthGlobalState` which takes the state and leave the other one without it but calling the new one.
2. (the more hacky IMO) have the GlobalState be "not required" by being variadic(`func Execute(...GlobalState)`)


In both cases sprinkled with some amount of comments on why this is needed.

Reply:
Thanks. Yeah, I guess the first is the way to go. :+1:

Reply:
Resolved by 96c1cca7.